pkg/apis/config/loader/loader.go

```diff
@@ -57,16 +49,14 @@ func LoadFromFile(filename string) (*config.ControllerConfiguration, error) {
 // Load takes a byte slice and de-serializes the contents into ControllerConfiguration object.
 // Encapsulates de-serialization without assuming the source is a file.
 func Load(data []byte) (*config.ControllerConfiguration, error) {
-	cfg := &config.ControllerConfiguration{}
```

Comment:
We already have some logic which generically takes care about decoding probably, see this [example](https://github.com/gardener/gardener-extension-provider-aws/blob/a7ae362d82c1652e70ccfebb856f291b1582e9a3/pkg/validator/serialization.go).